https://trac.macports.org/ticket/64115
OpenSSL 1.1 include path needs to be specified with higher precedence than ${prefix}/include
since openssl shim port now makes OpenSSL 3 include headers accessible from ${prefix}/include

--- ./cmake/ssl.cmake	2022-08-01 03:33:09.000000000 -0400
+++ ./cmake/ssl.cmake	2022-08-01 03:33:51.000000000 -0400
@@ -474,7 +474,7 @@
         SET(SSL_LIBRARIES ${SSL_LIBRARIES} ${LIBDL})
       ENDIF()
       MESSAGE(STATUS "SSL_LIBRARIES = ${SSL_LIBRARIES}")
-      INCLUDE_DIRECTORIES(SYSTEM ${OPENSSL_INCLUDE_DIR})
+      INCLUDE_DIRECTORIES(BEFORE ${OPENSSL_INCLUDE_DIR})
     ELSE()
       RESET_SSL_VARIABLES()
       FATAL_SSL_NOT_FOUND_ERROR(
@@ -542,8 +542,6 @@
         SET(HAVE_OPENSSL_DYLIB 1)
         IF(WITH_SSL STREQUAL "system")
           MESSAGE(STATUS "Using system OpenSSL from Homebrew")
-        ELSE()
-          SET(APPLE_WITH_CUSTOM_SSL 1)
         ENDIF()
       ENDIF()
     ENDIF(APPLE)
